aws/templates/id/id_spa.ftl

```diff
@@ -88,10 +89,42 @@
     }]
     
 [#function getSPAState occurrence]
+    [#local core = occurrence.Core]
+    [#local configuration = occurrence.Configuration]
+
+    [#assign cfId  = formatComponentCFDistributionId(core.Tier, core.Component, occurrence)]
+    [#assign cfName = formatComponentCFDistributionName(core.Tier, core.Component, occurrence)]
+
+    [#if configuration.Certificate.Configured && configuration.Certificate.Enabled ]
+            [#local certificateObject = getCertificateObject(configuration.Certificate!"", segmentId, segmentName) ]
+            [#local hostName = getHostName(certificateObject, core.Tier, core.Component, occurrence) ]
+            [#local fqdn = formatDomainName(hostName, certificateObject.Domain.Name)]
+    [#else]
+            [#local fqdn = getExistingReference(cfId,DNS_ATTRIBUTE_TYPE)]
+    [/#if]
+
     [#return
         {
-            "Resources" : {},
-            "Attributes" : {}
+            "Resources" : {
+                "cf" : {
+                    "Id" : cfId,
+                    "Name" : cfName
+                },
+                "cforiginspa" : {
```

Comment:
Should we be nesting these? I'd prefer we use hierarchy than compound names

Reply:
Perhaps it should be 2 cforigin resources with a diferent Id?